### PR TITLE
GTFS Support: Data Access and Parsing

### PR DESCRIPTION
**Pull request by chris-jpark**
# Issue

GTFS Support: Data Access and Parsing #3628 

Tests the ability to read and write GTFS files using the [just_gtfs](https://github.com/mapsme/just_gtfs/tree/cpp14) submodule. Imports the submodule into valhalla and creates a gurka test writing a GTFS folder that includes all necessary attributes used by valhalla. Tests that we can read and write GTFS files properly in valhalla.

### CODE REVIEW

**Review by nilsnolde — COMMENTED** [review]
thanks for the start @chris-jpark ! got a few comments:
- we need to take out all the stuff from the shortcuts PR (easiest is to revert the gtfs commits and change to a new branch based on master, the commit in that branch and open a new PR)
- I think we're ok without the `test/data/gtfs_sample` data which seems to be an altered toronto dataset, since we're able to roll our own gtfs data

 a bit more inline.

what do you think @kevinkreiser ?

File: test/gurka/test_gtfs.cc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  // write frequencies.txt
+  struct Frequency freqOne {
+    .trip_id = "bar", .start_time = Time(0, 0, 0), .end_time = Time(2, 0, 0), .headway_secs = 1800,
+    .exact_times = gtfs::FrequencyTripService::FrequencyBased,
+  };
+  feed.add_frequency(freqOne);
+  feed.write_frequencies("../test/data/gtfs_test/");
+
+  // make sure files are actually written
+  const auto& trips = feed.get_trips();
```
Comment (nilsnolde) [inline test/gurka/test_gtfs.cc:125]: you'll need to read the feed off disk, i.e. create a new feed from the directory path

File: test/gurka/test_gtfs.cc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+// test to write gtfs files
+TEST(GtfsExample, WriteGtfs) {
+
+  Feed feed;
+
+  // write agency.txt
+  struct Agency ttc {
+    .agency_id = "TTC", .agency_name = "Toronto Commission", .agency_url = "http://www.ttc.ca",
+    .agency_timezone = "America/Toronto"
+  };
```
Comment (nilsnolde) [inline test/gurka/test_gtfs.cc:28]: for now this test is fine I think. eventually we might want to encapsulate the gtfs writer in a function which we can call from single test suites where we have a bit control on e.g. coordinates for certain test maps and could actually integrate them with standard gurka maps even. anyways, future stuff. so far we're only testing `just_gtfs` actually, no valhalla api yet. but also there isn't any new valhalla api yet.

ideally we'd also make sure we produce valid gtfs data. you could run this tool on the `gtfs_test` dataset: https://github.com/MobilityData/gtfs-validator. seems to be well maintained. how we'll integrate that tool in the future to make sure all our (synthetic) test gtfs data is eventually valid, not sure yet. if we take this current dataset for the time being, let's validate it manually first before moving on.

**Comment by kevinkreiser** [comment]
@chris-jpark lets close this in favor of the other one you opened up

**Pull request closed without merging**